sdks/rust/src/protocol/mod.rs

```diff
@@ -16,6 +16,162 @@ pub mod payload;
 pub mod pike;
 pub mod state;
 
+use std::error::Error;
+
+#[cfg(not(target_arch = "wasm32"))]
+use transact::signing::{hash::HashSigner, Signer};
+
 pub const ADMINISTRATORS_SETTING_ADDRESS: &str =
     "000000a87cb5eafdcca6a814e4add97c4b517d3c530c2f44b31d18e3b0c44298fc1c14";
 pub const ADMINISTRATORS_SETTING_KEY: &str = "sawtooth.swa.administrators";
+
+pub const NAMESPACE_REGISTRY_ADDRESS_PREFIX: &str = "00ec00";
+pub const CONTRACT_REGISTRY_ADDRESS_PREFIX: &str = "00ec01";
+pub const CONTRACT_ADDRESS_PREFIX: &str = "00ec02";
+pub const SMART_PERMISSION_ADDRESS_PREFIX: &str = "00ec03";
+pub const AGENT_ADDRESS_PREFIX: &str = "cad11d00";
+pub const ORG_ADDRESS_PREFIX: &str = "cad11d01";
+
+/// Compute a state address for a given namespace registry.
+///
+/// # Arguments
+///
+/// * `namespace` - the address prefix for this namespace
+#[cfg(not(target_arch = "wasm32"))]
+pub fn compute_namespace_registry_address(namespace: &str) -> Result<Vec<u8>, AddressingError> {
```

Comment:
Do we have a current use-case for these being pub? (Both the functions and const values here.) If not, suggest we restrict them with pub(crate) so they aren't part of the public API we support.

Reply:
The scabbard CLI in splinter needs the contract registry prefix and needs to compute contract addresses to show/list contracts.